### DISCUSSION TITLE
[v3.4.9] v-data-iterator doesn't correctly type items

### DISCUSSION THREAD

**Discussion by douglasg14b** [vuetifyjs/vuetify#19285] [Q&A]
The `items` array of data iterator is always typed with `DataIteratorItem<any>[]` and doesn't appear to type to the input items.

Is this a bug? How do I correct this behavior?

### Typed Input:

![image](https://github.com/vuetifyjs/vuetify/assets/1400380/75e8e94a-c5bd-45e4-887e-9172025e850b)

## Untyped default slot items:

![image](https://github.com/vuetifyjs/vuetify/assets/1400380/5f6c1db2-f7b4-4aeb-80a5-ec95efca9040)

```vue
                    <v-data-iterator :items="creatives">
                        <template v-slot="{ items }">
                            <v-card v-for="item in items" :key="item.raw.id">
                                <div class="d-flex flex-no-wrap justify-space-between">
                                    <v-card-title class="text-h5">
                                        {{ item.raw.name }}
                                    </v-card-title>
                                </div>
                            </v-card>
                        </template>
                    </v-data-iterator>
```

**Comment by QimatLuo** [comment]
Same issue here.

**Reply by fgael** [reply]
Hello, it have been fix in vuetify 3.5.7 https://github.com/vuetifyjs/vuetify/releases/tag/v3.5.7
